clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -1460,7 +1486,18 @@ void checkExprLifetime(Sema &SemaRef, const AssignedEntity &Entity,
 
   checkExprLifetimeImpl(SemaRef, /*InitEntity=*/nullptr,
                         /*ExtendingEntity=*/nullptr, LK_Assignment, &Entity,
-                        Init);
+                        /*CapEntity=*/nullptr, Init);
+}
+
+void checkCaptureByLifetime(Sema &SemaRef, const CapturingEntity &Entity,
+                            Expr *Init) {
+  if (!SemaRef.getDiagnostics().isIgnored(
+          diag::warn_dangling_reference_captured, SourceLocation()))
```

Comment:
we should do it for `warn_dangling_reference_captured_by_unknown` as well?

Reply:
I was inclining towards using only one warning here. But happy to add both as well.
Analysis is off if both of the warnings are ignored.